projects/accredited-programmes-and-oasys/src/main/kotlin/uk/gov/justice/digital/hmpps/controller/RiskPrediction.kt

```diff
@@ -0,0 +1,95 @@
+package uk.gov.justice.digital.hmpps.controller
+
+import uk.gov.justice.digital.hmpps.controller.ScoreLevel.Companion.of
+import java.math.BigDecimal
+import java.time.LocalDateTime
+
+data class RiskPrediction(
+    val completedDate: LocalDateTime? = null,
+    val assessmentStatus: String? = null,
+    val groupReconvictionScore: YearPredictor? = null,
+    val violencePredictorScore: YearPredictor? = null,
+    val generalPredictorScore: YearPredictor? = null,
+    val riskOfSeriousRecidivismScore: RsrPredictor? = null,
+    val sexualPredictorScore: SexualPredictor? = null,
+)
+
+data class YearPredictor(
+    val oneYear: BigDecimal? = null,
+    val twoYears: BigDecimal? = null,
+    val scoreLevel: ScoreLevel? = null,
+) {
+    companion object {
+        fun of(oneYear: BigDecimal?, twoYears: BigDecimal?, scoreLevel: String?): YearPredictor? =
```

Comment:
Are you using any of these `of` functions?

Reply:
Are they called implicitly by Jackson or am I missing something?  Probably the latter - it's Friday afternoon 😄

Reply:
Doh sorry just spotted the extension function!